### PR TITLE
feat: Workflow Editing Modal

### PR DESCRIPTION
**Pull request by nianxh-milesight**
image
Workflow Editing Modal

### CODE REVIEW

File: apps/web/src/pages/workflow/components/edit-modal/index.tsx [hunk 1]
```diff
@@ -1 +1,59 @@
...
+
+    const handleCancel = useMemoizedFn(() => {
+        reset();
+        onCancel?.();
+    });
+
+    return (
+        <Modal
+            visible={visible}
+            title={getIntlText('common.label.edit_workflow_modal')}
```
Comment (Jimco) [inline apps/web/src/pages/workflow/components/edit-modal/index.tsx:45]: The title should be changed to [Add Workflow] when this is a add modal.

File: packages/locales/src/lang/en/global.json [hunk 2]
```diff
@@ -158,5 +158,7 @@
     "common.label.enable_status": "Enable Status",
     "common.label.update_time": "Update Time",
     "common.label.back": "Back",
-    "common.label.create": "Create"
+    "common.label.create": "Create",
+    "common.label.edit_workflow_modal": "Edit Workflow",
+    "common.label.workflow_name": "Workflow Name"
```
Comment (Jimco) [inline packages/locales/src/lang/en/global.json:163]: The above two lines are not generic text and should be placed in the business module (`workflow.json`).

File: apps/web/src/pages/workflow/components/edit-modal/index.tsx [hunk 3]
```diff
@@ -1 +1,59 @@
...
+import { useMemoizedFn } from 'ahooks';
+import cls from 'classnames';
+import { useForm, Controller, type SubmitHandler } from 'react-hook-form';
+import { useI18n } from '@milesight/shared/src/hooks';
+import { Modal, toast, type ModalProps } from '@milesight/shared/src/components';
+import useEditFormItems, { type FormDataProps } from './hook/useEditFormItems';
+
+interface Props extends Omit<ModalProps, 'onOk'> {
+    /** error callback */
+    onError?: (err: any) => void;
```
Comment (Jimco) [inline apps/web/src/pages/workflow/components/edit-modal/index.tsx:11]: The `onError` prop is not used in component ?

File: apps/web/src/pages/workflow/components/edit-modal/index.tsx [hunk 4]
```diff
@@ -1 +1,59 @@
...
+
+    /** success callback */
+    onSuccess?: (params: FormDataProps) => void;
```
Comment (Jimco) [inline apps/web/src/pages/workflow/components/edit-modal/index.tsx:14]: There are no API requests in the component, so the callback function should not be named `onSuccess`. Instead, `onConfirm` can be used.

File: apps/web/src/pages/workflow/components/edit-modal/index.tsx [hunk 5]
```diff
@@ -1 +1,59 @@
...
+
+    /** table-row item */
+    data: FormDataProps;
```
Comment (Jimco) [inline apps/web/src/pages/workflow/components/edit-modal/index.tsx:17]: The component needs to support adding scenarios, so `data` should be an optional prop.

File: apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import { TextField } from '@mui/material';
+import { useI18n } from '@milesight/shared/src/hooks';
+import { checkRequired } from '@milesight/shared/src/utils/validators';
+
+/**
+ * type of dataSource
+ */
+export type FormDataProps = {
+    name: string;
+    remark: string;
```
Comment (Jimco) [inline apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx:12]: The `remark` prop is  optional.

File: apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    const { getIntlText } = useI18n();
+    const { name, remark } = initialData;
+    const formItems = useMemo(() => {
+        const result: ControllerProps<FormDataProps>[] = [];
+        result.push(
+            {
+                name: 'name',
+                defaultValue: name,
+                rules: {
+                    validate: { checkRequired: checkRequired() },
```
Comment (Jimco) [inline apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx:25]: Missing text length verification.

File: apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                            value={value}
+                            onChange={onChange}
+                        />
+                    );
+                },
+            },
+            {
+                name: 'remark',
+                defaultValue: remark,
+                render({ field: { onChange, value }, fieldState: { error } }) {
```
Comment (Jimco) [inline apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx:45]: Should add the min&max text length verification.

File: apps/web/src/pages/workflow/components/edit-modal/index.tsx (hunk 4 above)
Comment (nianxh-milesight) [reply apps/web/src/pages/workflow/components/edit-modal/index.tsx:14]: Change to onConfirm method

File: apps/web/src/pages/workflow/components/edit-modal/index.tsx (hunk 3 above)
Comment (nianxh-milesight) [reply apps/web/src/pages/workflow/components/edit-modal/index.tsx:11]: onOk={handleSubmit(onSubmit, onError)}

File: apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx (hunk 8 above)
Comment (nianxh-milesight) [reply apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx:45]: maxLength: 1000 & not required

File: apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx (hunk 7 above)
Comment (nianxh-milesight) [reply apps/web/src/pages/workflow/components/edit-modal/hook/useEditFormItems.tsx:25]: maxLength: 50

**Pull request merged**